text/0026-rest-api/rest-api.yaml

```diff
@@ -0,0 +1,707 @@
+      tags:
+        - node
+      summary: Returns information about the node.
+      description: Returns information about the node.
+      operationId: get-info
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/GetInfoResponse'
+              examples:
+                Example 1:
+                  $ref: '#/components/examples/get-info-response-example-1'
+  /tips:
+    get:
+      tags:
+        - tangle
+      summary: Returns two non-lazy tips.
+      description: >-
+        Returns two non-lazy tips. In case the node can only provide one tip,
+        tip1 and tip2 are identical.
+      operationId: get-tips
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/GetTipsResponse'
+              examples:
+                Example 1:
+                  $ref: '#/components/examples/get-tips-response-example-1'
+                Example 2:
+                  $ref: '#/components/examples/get-tips-response-example-2'
+  '/messages/by-id/{messageId}':
+    get:
+      tags:
+        - messages
+      summary: Find a message by its identifier.
+      description: >-
+        Find a message by its identifer. This endpoint returns the given message
+        including payload.
+      operationId: get-messages-by-id
+      parameters:
+        - in: path
+          name: messageId
+          schema:
+            type: string
+          required: true
+          description: Identifier of the message.
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GetMessagesByIdResponse'
+              examples:
+                Example 1:
+                  $ref: '#/components/examples/get-messages-by-id-response-example-1'
+                Example 2:
+                  $ref: '#/components/examples/get-messages-by-id-response-example-2'
+        '400':
+          description: unsuccessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/invalid-data-provided-response-example'
+        '404':
+          description: unsuccessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/can-not-find-data-response-example'
+  '/messages/by-index/{index}':
```

Comment:
What do you think about making `by-index` and other options parameters instead? So it will be `GET /messages?index={index}`, and vice versa.